cmd/icingadb/main.go

```diff
@@ -67,8 +72,38 @@ func run() int {
 		}
 	}
 
-	if err := icingadb.CheckSchema(context.Background(), db); err != nil {
-		logger.Fatalf("%+v", err)
+	{
+		err := retry.WithBackoff(
+			context.Background(),
+			func(ctx context.Context) (err error) {
+				return database.AutoUpgradeSchema(
+					context.Background(), db, cmd.Config.Database.Database, "icingadb_schema", "version", "timestamp",
+					map[string]database.SchemaData{
+						database.MySQL: {
+							Schema: []string{mysql.Schema},
+							Upgrades: []database.SchemaUpgrade{
+								{Version: "3", DDL: []string{mysql.Upgrade100}},
+								{Version: "4", DDL: []string{mysql.Upgrade111}},
+								{Version: "5", DDL: []string{mysql.Upgrade120}},
+							},
+						},
+						database.PostgreSQL: {
+							Schema: []string{pgsql.Schema},
+							Upgrades: []database.SchemaUpgrade{
+								{Version: "2", DDL: []string{pgsql.Upgrade111}},
+								{Version: "3", DDL: []string{pgsql.Upgrade120}},
+							},
+						},
+					},
+				)
+			},
+			retry.Retryable,
+			backoff.NewExponentialWithJitter(128*time.Millisecond, time.Minute),
+			db.GetDefaultRetrySettings(),
```

Comment:
Please correct me if I'm wrong, but this would force (multiple) schema upgrades to be performed in five minutes. If this takes more time, the user ends up in a very messy situation, potentially don't even knowing how they got there.

There were already long running schema upgrades in the past, <https://icinga.com/docs/icinga-db/latest/doc/04-Upgrading/#upgrading-to-icinga-db-v120>.